drv/sidecar-front-io/src/transceivers.rs

```diff
@@ -77,6 +90,144 @@ impl FpgaPortMasks {
     }
 }
 
+/// Represents a single logical port (0-31)
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd)]
+pub struct LogicalPort(pub u8);
+impl LogicalPort {
+    pub fn as_mask(&self) -> LogicalPortMask {
+        LogicalPortMask(1 << self.0)
+    }
+
+    pub fn get_physical_location(&self) -> PortLocation {
```

Comment:
This could delegate to `PortLocation::from(*self)`, instead of looking into `PORT_MAP` itself.